### PR TITLE
Remove usage of deprecated CDI API.

### LINKED ISSUE #103: Usage of deprecated methods from CDI API
`TestContainer` uses `BeanManager.fireEvent` which has been removed in latest CDI API (4.0.0.Alpha1).
We will need to adapt that.

However, note that in order to pass tests, we will also need a Arq. core update for similar reasons - see https://github.com/arquillian/arquillian-core/pull/339
Weld and CDI API don't need to be updated immediately (can't actually because of circular dep).

I will send a draft PR that captures what we can fix and which should be updated once Arq. core is out.

### PR DESCRIPTION
**Pull request by manovotn**
Fixes #103 

Latest CDI API (4.0.0.Alpha1) removed some long deprecated API.
Therefore, using this container will cause a failure with `NoSuchMethod` - spotted as part of https://github.com/weld/core/pull/2383

~~Not to be merged until there is a release of Arq. core that contains https://github.com/arquillian/arquillian-core/pull/339~~

### CODE REVIEW

**Comment by manovotn** [comment]
Since Travis is deprecated, I have also added the GH Actions template to this PR but apparently it won't trigger any runs until the file has been merged into master. The build itself works locally so I expect no issue with that.

**Comment by manovotn** [comment]
@starksm64 @bartoszmajsak I could use a review here and eventually a release on this project :)

File: .github/workflows/ci-actions.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Arquillian Weld Container CI
```
Comment (bartoszmajsak) [inline .github/workflows/ci-actions.yml:1]: Any particular reason for GH action vs Circle CI? It's good to see `.travis.yml` problem being solved*, I just wonder if we should strive for having a unified CI view and maintenance for the whole org (with the downside being shared quotas for example) or leave freedom to choose for every project.


*  on a personal level I'm a bit sad about what happened to this milestone in CI space after acquisition though :(

**Review by bartoszmajsak — COMMENTED** [review]
LGTM - just one question with regards to CI

File: .github/workflows/ci-actions.yml (hunk 1 above)
Comment (manovotn) [reply .github/workflows/ci-actions.yml:1]: Good point, I didn't even think about it. I used GH actions throughout Weld projects so I automatically added it here as well.

I am only familiar with Travis (whose free version was a joke last time I checked) and with GH Actions which has fairly astonishing throughput and limits for free usage. So I am biased towards GH actions :-)
I don't know CircleCI and what ups/downs it has but if you prefer it, I can change it - your call.

Comment (manovotn) [reply .github/workflows/ci-actions.yml:1]: I checked the how-to setup for CircleCI but it seems it requires admin to set it up on their website which I cannot do either way. But maybe there is another way I missed.

@bartoszmajsak let me know how you want to proceed - if you are fine with GH actions, I think we can merge it as-is. If you want CircleCI, I can withdraw GH actions changes from this PR.

Comment (bartoszmajsak) [reply .github/workflows/ci-actions.yml:1]: I think we can stick with it if that works for you. Gonna push the release buttons today :)

**Pull request merged**